inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -423,7 +530,7 @@ void ngfunction_2_irv10(pugi::xml_document& doc,
 
         const auto data_attr_size =
```

Comment:
@jdanieck 
Can we switch to:
```
const bool no_attributes = data.attributes().begin() == data.attributes().end();
```
This will skip unnecessary traverse whole list of attributes.

Reply:
@pelszkow intent of this code is to not introduce <data> node to IR if there are no attributes, so if std::distance doesn't do the same under the hood we can switch to the code you proposed.

@sdurawa why are we skipping this check for TensorIterator?

Reply:
TensorIterator does not have attributes in "data" section. For TI "data" layer (passed to XmlSerializer) is renamed to "body" layer and filled with nGraph function. This is the reason of not removing an empty layer for TI.

Reply:
Skipping for TI is removed.